src/components/Canvas/Canvas.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+    forwardRef,
+    ReactNode,
+    useEffect,
+    useImperativeHandle,
+    useRef,
+    useState
+} from "react";
+import { CanvasContext2d } from "../../hooks/useContext2d";
+
+type CanvasProps = {
+    height: number;
+    width: number;
+    children?: ReactNode;
+};
+
+export const Canvas = forwardRef<
+    HTMLCanvasElement,
+    CanvasProps
+>((
+    { height, width, children },
+    ref
+) => {
+    const [context2d, setContext2d] = useState<CanvasRenderingContext2D | null>(
+        null
+    );
+    const canvasRef = useRef<HTMLCanvasElement | null>(null);
+
+    useImperativeHandle(ref, () => canvasRef.current!);
```

Comment:
I haven't seen this used before. I read the [react documentation ](https://react.dev/reference/react/useImperativeHandle) but I don't quite get what this affords you.

Reply:
I want to use the ref that I'm forwarding in the element that is forwarding it. I'm not sure if there's a better way to do that.

Reply:
Based on some quick research it looks like this might be the common way of doing it. I'm not sure if there's a better way, but it makes sense to me.